tests/core/test_statistics.cpp

```diff
@@ -0,0 +1,124 @@
+#include "cle.hpp"
+#include <array>
+#include <gtest/gtest.h>
+#include <cmath>
+
+class TestStatisticsOfLabelledPixels : public ::testing::TestWithParam<std::string>
+{
+protected:
+    std::array<float, 3 * 3 * 1> intensity = {0, 1, 2, 1, 2, 3, 2, 3, 4};
+    std::array<float, 3 * 3 * 1> labels = {1, 1, 2, 1, 2, 2, 3, 3, 3};
+    std::array<float, 36 * 3 * 1> reference = {
+        // IDENTIFIER(0)
+        1, 2, 3,
+        // BOUNDING_BOX_X(1)
+        0, 1, 0,
+        // BOUNDING_BOX_Y(2)
+        0, 0, 2,
+        // ... [rest of the reference data]
+    };
+    std::array<float, 36 * 3 * 1> output;
+};
+
+TEST_P(TestStatisticsOfLabelledPixels, execute)
+{
+    std::string param = GetParam();
+    cle::BackendManager::getInstance().setBackend(param);
+    auto device = cle::BackendManager::getInstance().getBackend().getDevice("", "all");
+    device->setWaitToFinish(true);
+
+    auto gpu_intensity = cle::Array::create(3, 3, 1, 3, cle::dType::FLOAT, cle::mType::BUFFER, device);
+    gpu_intensity->write(intensity.data());
+    auto gpu_labels = cle::Array::create(3, 3, 1, 3, cle::dType::FLOAT, cle::mType::BUFFER, device);
+    gpu_labels->write(labels.data());
+
+    auto result = cle::tier1::statistics_of_labelled_pixels_func(device, gpu_intensity, gpu_labels, nullptr);
```

Comment:
This function does obviously not exist in tier1. How could I call it? Or shall I put it in tier8?

Reply:
you need to include your `.hpp` folder, here `statistics.hpp` in the test and you can call it like `cle::statistics_of_labelled_pixels_func`

It is ok to keep the function in an `hpp` and outside of any `tiers` for now. For this I was aiming to something similar to the `transform`. Having a `core` code hidden in charge of the heavy code and then make 1 function available in the correct tiers later on.